lib/nerves_network/resolvconf.ex

```diff
@@ -152,21 +158,29 @@ defmodule Nerves.Network.Resolvconf do
     new_ifentry = state.ifmap
                     |> Map.get(ifname, %{})
                     |> Map.merge(%{:static_domains => domains})
+
     state = %{state | ifmap: Map.put(state.ifmap, ifname, new_ifentry)}
+
     Logger.debug fn -> "#{__MODULE__}: handle_call: :set_static_domains new_state = #{inspect state}" end
     write_resolvconf(state)
     {:reply, :ok, state}
   end
 
   def handle_call({:set_static_nameservers, ifname, servers}, _from, state) do
     Logger.debug fn -> "#{__MODULE__}: handle_call: :set_static_nameservers old_state = #{inspect state}" end
-    new_ifentry = state.ifmap
-                    |> Map.get(ifname, %{})
-                    |> Map.merge(%{:static_nameservers => servers})
-    state = %{state | ifmap: Map.put(state.ifmap, ifname, new_ifentry)}
-    Logger.debug fn -> "#{__MODULE__}: handle_call: :set_static_domains new_state = #{inspect state}" end
-    write_resolvconf(state)
-    {:reply, :ok, state}
+    #Remove the old static_nameservers from the nameservers superposition
+    ifmap = state.ifmap |> Map.get(ifname, %{})
+
+    current_static_nameservers = Map.get(ifmap, :static_nameservers, [])
+    current_nameservers        = Map.get(ifmap, :nameservers, [])
+
+    nameservers = current_nameservers -- current_static_nameservers
```

Comment:
What if we stored these as MapSets instead? They should be sets and have unique elements. We are also trying to do set operations on them like difference.

Reply:
Here are the docs https://hexdocs.pm/elixir/MapSet.html#content

It doesn't say when they were added but I think it was after Elixir 1.0